### PR TITLE
[REVIEW] add spdlog as a conda dependency

### LINKED ISSUE #359: [FEA] Include spdlog headers in conda package
**Is your feature request related to a problem? Please describe.**

The librmm conda package doesn't ship the spdlog headers. This is a problem for anyone who wants to use the `logging_resource_adaptor`.

The spdlog headers currently live in the `thirdparty` dir https://github.com/rapidsai/rmm/tree/branch-0.14/thirdparty/spdlog

**Describe the solution you'd like**
Ship the `spdlog` headers in a way that RMM conda users can access them.

### PR DESCRIPTION
**Pull request by rongou**
@jrhemstad @harrism 

Fixes #359

### CODE REVIEW

**Comment by GPUtester** [comment]
Please update the changelog in order to start CI tests.

View the gpuCI docs [here](https://docs.rapids.ai/gpuci#open-pr-or-update-pr).

**Comment by jakirkham** [comment]
Should we add to `host` as well so the same version gets used in both cases?

**Comment by jrhemstad** [comment]
> Should we add to `host` as well so the same version gets used in both cases?

host?

**Comment by rongou** [comment]
@jakirkham makes sense. Done.

File: conda/recipes/librmm/meta.yaml [hunk 1]
```diff
@@ -27,8 +27,10 @@ requirements:
     - cmake >=3.12.4
   host:
     - cudatoolkit {{ cuda_version }}.*
+    - spdlog >=1.4.2
   run:
     - {{ pin_compatible('cudatoolkit', max_pin='x.x') }}
+    - spdlog >=1.4.2
```
Comment (jakirkham) [inline conda/recipes/librmm/meta.yaml:33]: Would suggest tweaking this a little bit. Should make it easier to bump the version as needed.

```suggestion
  host:
    - cudatoolkit {{ cuda_version }}.*
    - spdlog 1.4.2
  run:
    - {{ pin_compatible('cudatoolkit', max_pin='x.x') }}
    - {{ pin_compatible('spdlog', max_pin='x.x') }}
```

**Comment by jakirkham** [comment]
> > Should we add to `host` as well so the same version gets used in both cases?
> 
> host?

Yeah the terminology comes [from autoconf]( https://www.gnu.org/software/autoconf/manual/autoconf-2.69/html_node/Hosts-and-Cross_002dCompilation.html ). Though we are not actually cross-compiling. So `host` here just means the dependencies available for linking against during the build.

File: conda/recipes/librmm/meta.yaml (hunk 1 above)
Comment (rongou) [reply conda/recipes/librmm/meta.yaml:33]: Wouldn't this make it stuck at 1.4.2 until we update it? Anyway, spdlog is a header-only library and we are not using any fancy features, the versioning probably doesn't need to be that strict. Perhaps we just leave out the version altogether?

**Comment by rongou** [comment]
@jakirkham @jrhemstad can this be merged now?

**Pull request merged**